### ISSUE TITLE
[Q] (Your Question?)where the BTT mini UPS v1.0 has to be connected to the TFT35-E3 V3.0? 

### ISSUE THREAD

**Issue by killrob** [bigtreetech/BIGTREETECH-TouchScreenFirmware#1458]
Hi, i have a TFT35 E3 V3.0 and a BTT mini UPS V1.0, where i have to connect the mini UPS to the TFT?
in UART connectors or in PS-ON connector?
`
#### BTT UPS Support
# Enable BTT UPS
#
# Options: [enable: 1, disable: 0]
#
btt_mini_ups:0`

this is the code in config.ini, but you don't say where the mini UPS has to be connected in TFT35
thanks

**Comment by oldman4U** [comment]
Guess not many here use this hardware because resuming a print works out of the box using the TFT alone.

**Comment by killrob** [comment]
> 
> 
> Guess not many here use this hardware because resuming a print works out of the box using the TFT alone.

Sure it works, but when there is a blackout the nozzle stops there and the print will be ruined, i would like to use the mini UPS to raise Z at least 10mm, but in touch mode doesn't work, i have still to try in simulator mode.

**Comment by oldman4U** [comment]
It is not supported. Please see #1193

**Comment by mpcolman** [comment]
That's a big issue... I have the UPS module to install it and Im breakfast with this throuble........

**Comment by killrob** [comment]
well.. i think i will uninstall it until BTT release some new firmware for TFT that makes mini UPS work connected to it, as the Smart filament sensor do. I'm not intersted to use it in Marlin mode.

**Comment by oldman4U** [comment]
@berrenger was able to use it with the Touchscreen but also with limitations. You have to watch the video he provided in #1193.

**Comment by killrob** [comment]
i saw the video but he uses a BTT SKR V1.4, i use a mini E3V2.0. 
But no probs i will wait for BTT ;)

**Comment by oldman4U** [comment]
So the problem is in the mainboard and not the TFT?

You know that the firmware is made by users like you and me, so waiting for BTT to implement it can take a long time.

**Comment by killrob** [comment]
not really, the problem is that i would like to use mini UPS in touch mode with my TFT35E3V3.0 and the SKR mini E3V2.0, but i'm not so skilled in programming like our friend @berrenger, so i will wait for someone who will do the job, meanwhile i will try to take a look on source code of the TFT35 firmware to try to understand if is a job that i can do by myself.

**Comment by oldman4U** [comment]
Maybe @berrenger can help. He seems to be a smart guy.

killrob <notifications@github.com> schrieb am Sa. 2. Jän. 2021 um 22:15:

**Comment by killrob** [comment]
maybe, but i will not ask cause i don't want to bore him with a problem that is mine. I will take this chance to become a little more skilled in firmware programming 😄

**Comment by oldman4U** [comment]
But we help each other here and the benefit is that you meet new persons and that you learn a lot. I would also be interested how it works and once I know, we can add this information to the ReadMe. So hopefully berrenger shows up here and helps us;-)

**Comment by killrob** [comment]
today i have do nothing cause a blackout for quite all day, but i agree with you, once i find the solution, if i find it, i will post here, so can be added to the readme.

**Comment by berrenger** [comment]
Hi there. I stopped using "btt mini ups".  I constantly encounter different problems, so I stopped using it.  I hope you can create a synergy and solve all the problems.

**Comment by oldman4U** [comment]
Thank you. This would be great.

killrob <notifications@github.com> schrieb am So. 3. Jän. 2021 um 19:50:

**Comment by killrob** [comment]
> 
> 
> Hi there. I stopped using "btt mini ups". I constantly encounter different problems, so I stopped using it. I hope you can create a synergy and solve all the problems.

hi @berrenger thank you for reply, are you using something else? perhaps the MKS mini ups?

**Comment by gordo3di** [comment]
@killrob I've done a lot of testing with the BTT UPS 12v v1.0 module, SKR v1.4 turbo, PLR and Marlin 2.0.5.x. After exhaustive efforts here are my findings:

- BTT module must be connected to the SKR board. I used the "PWRDET" pin recommended. Its basically pin P1.0 and just turn on all the stuff in Marlin.
-
- I could not get the z-raise to work when the power loss is detected. The one time I managed to get it working (magically) the z did raise but the PLR didn't recognize so it was pointless... Still more testing to do...

- PLR works in Marlin emulation mode and it will go back to the correct position. Problem is that the hotend didn't wait to heat before moving so if it cooled on the print it will sometimes get stuck. Otherwise works correctly.

- PLR works in TFT mode but only in the traditional sense (save the recovery per layer and reprint the layer from the start). The firmware for the TFT needs to have the PLR detection implemented. On another note, the hotend does heat before moving though.

Hope this helps. I'll continue to work on it.
